cloud/terraform/aws/installation.md

```diff
@@ -0,0 +1,144 @@
+
+## EKS Cluster Bootstrap
+
+1. Navigate to the `eks` directory. 
+
+2. Edit `terraform.tfvars` to configure your infrastructure. Variables are defined in the `variables.tf` file. By default, the system deploys an EKS cluster in the US West (Oregon) region (us-west-2) spanning two availability zones.
+
+3. Run the following commands:
+
+   ```bash
+   terraform init
+   ```
+ 
+   ```bash
+   terraform apply
+   ```
+
+   You are prompted to apply the changes. 
+
+4. Append the result of `terraform output` to the `../dependencies/terraform.tfvars` file in the `dependencies` directory: 
+
+   ```bash
+   terraform output >> ../dependencies/terraform.tfvars
+   ```
+
+## Liferay Infrastructure Bootstrap
+
+1. Navigate to the `dependencies` directory.
+
+2. Update the `terraform.tfvars` file to configure your infrastructure. Variables are defined in `variables.tf` file. If you followed [EKS Cluster Bootstrap](#eks-cluster-bootstrap), this file is already populated.
+
+3. Run the following commands:
+
+   ```bash
+   terraform init
+   ```
+
+   ```bash
+   terraform apply
+   ```
+
+   You are prompted to apply the changes. 
+
+<!-- Just a sanity check on the above two sections. It sounds like there are two `terraform.tfvars` files in two different directories: `eks` and `dependencies`. In EKS Cluster Bootstrap, you have the reader editing `terraform.tfvars` in `eks`, and then append the result of a command into `../dependencies/terraform.tfvars`. In Liferay Infrastructure Bootstrap, you have the reader edit only `dependencies/terraform.tfvars`, but the instructions (wordsmithed) correspond to the instructions above for the `eks/terraform.tfvars` file. -->
+
+## Helm Setup
+
+To use Helm you must use the `aws` CLI to set up `kubectl`. 
+
+1. Navigate to the `dependencies` directory.
+
+2. Run the command below: 
+
+   ```bash
+   aws eks --region $(terraform output -raw region) update-kubeconfig --name $(terraform output -raw cluster_name)
+   ```
+
+3. Test that `kubectl cluster-info` works.
+
+## Helm Chart Deployment
+
+The chart expects a Kubernetes secret called `managed-service-details` in the deployment namespace containing the following data:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+    name: managed-service-details
+data:
+    OPENSEARCH_ENDPOINT: ""
+    OPENSEARCH_PASSWORD: ""
+    OPENSEARCH_USERNAME: ""
+    DATABASE_ENDPOINT: ""
+    DATABASE_PASSWORD: ""
+    DATABASE_PORT: ""
+    DATABASE_USERNAME: ""
+    S3_BUCKET_ID: ""
+    S3_BUCKET_REGION: ""
+```
+
+This secret was created when you initialized and applied the Terraform configuration. If you skipped that, it must be provided manually.
+
+<!-- Which step above initialized this file? I made a guess; please verify my guess was right. :-)
+     Also, the instructions above provide no use case for skipping a step. How would this happen? 
+-->
```

Comment:
This `Secret` resource would be created within the Liferay Infrastructure Bootstrap procedure. It would be generated locally by the `terraform init` command and then applied to the cluster via `terraform apply`.

The terraform resource is declared here: https://github.com/brianchandotcom/liferay-portal/pull/160633/files#diff-777fac9b0d21c19c4c0607dda734e3fdc72fb5187e15467246efec87c7950ddbR199-R216

Reply:
I'm not sure of an actual use case for skipping the Liferay Infrastructure Bootstrap, but my understanding is that this information enables a user to deploy the Helm chart without executing the bootstrap procedure. I'm curious if @rotty3000 has any input.

Agree that it's slightly unclear as is.

Reply:
@sez11a - @rotty3000 says the use case is basically if the customer already has an existing EKS cluster to deploy into and needs to connect their Liferay to the cloud provider's managed services stack that they're bringing along. It's a BYOB approach. Does that help?

Reply:
@jaredgorski But in both sections you have the same instructions where you run `terraform init` and `terraform apply`. If the secret resource is created during that procedure when you run those commands, you'll never skip that step. That's why I'm confused.

Reply:
@sez11a There are some users who may already have their own infrastructure already running on EKS and prefer to DIY the bootstrap rather than running the Terraform auto-configuration. However, their Liferay instance may still need external search, database, and object storage. Those users need to know that the Helm chart needs this `managed-service-details` resource in order to know how to configure Liferay to talk to Amazon's managed OpenSearch, database, or object storage solutions.

If need be, we can set up a short call to smooth everything out. LMK